### ISSUE TITLE
Add a property to the spring-boot-starter allowing custom document format registry.

### ISSUE THREAD

**Issue by sbraconnier** [sbraconnier/jodconverter#94]
See #92 

This change will allow us to revert #93.

**Comment by EugenMayer** [comment]
@sbraconnier i would at least throw up the discussion if that actually fixes #93. If you have a potential user of jodconverter, i do not expect him to know anything about Filters, how to apply options and that such an option exists. You have seen that live by me looking for that.

IMHO I would love to see #94 happen, but in addition, add a new documenType to the current json ( html:embedded ) and a new jodoption "embedd images" which then just remaps any html type on that automatically, if the use has set the flag. I think that is a separate discussion, for me personally #94 would totally do it - but i think we might have other users not being even aware of how to do that?

**Comment by sbraconnier** [comment]
Sorry for the long delay. My solution took more time than I thought to implement. So the idea is that now the `jodconverter-spring-boot-starter` module allow you to replace the whole document formats registry through a custom JSON. But as you said, it could be tricky for a user to do that if its goal is just to add a single option, like the `EmbedImages` one. So, I also added the ability to customize existing document format options through properties that can be set in the application.properties (or yml) configuration file. The following application.yml:

```yml
jodconverter:
  local:
    enabled: true
    format-options:
      html:
        store:
          TEXT:
            FilterOptions: EmbedImages
```

will add the `"FilterOptions": "EmbedImages"` to the existing HTML format definition loaded from the `document-formats.json` file.

Now I know the documentation is not so great. I will have to improve it for this solution to be understood by jodconverter users!

**Comment by EugenMayer** [comment]
That is, IMHO, the best choice you could make. Perfect for all sides. 

I expect that the old ways, not embedding, will be the default, right? ( it makes sense for now ). You might consider a migration time for this, with a major release or something, when the embed image is battle proven.

We are using it in production for 1 month now - no issues, also with bigger documents and such.

Happy to see this getting done! Any ETA on a release including this? Thanks!

**Comment by sbraconnier** [comment]
You are right, not embedding the image is the default (#93 has been reverted).

I hope to be able to create the next release in a week or so. I'm not sure if it will be a 4.3.0 or a 4.2.1 version since there will be no breaking changes if I remember correctly.

**Comment by EugenMayer** [comment]
@sbraconnier wanted you to know i finally implemented this and switch to the official 4.2.2 library and it worked out without any issues. I added a test for it https://github.com/EugenMayer/officeconverter/blob/master/src/test/java/de/kontextwork/converter/web/ConversionControllerTest.java#L79 and configured it like expected

https://github.com/EugenMayer/officeconverter/blob/master/src/main/resources/application.yml#L22

I am not sure how i would re-adjust his configuration during a test to test the negative test, but at least that one will do for me.

Thank you a lot and sorry for the delay :)
